tests/integration/test.attachments.js

```diff
@@ -1518,6 +1519,53 @@ adapters.forEach(function (adapter) {
         });
     });
 
+    it('Test getAttachment with specific rev', function () {
+      var db = new PouchDB(dbs.name);
+
+      var doc = {
+        _id: 'a',
+      };
+      var rev1;
+      var rev2;
+      var rev3;
+      return db.put(doc).then(function (res) {
+        doc._rev = rev1 = res.rev;
+        doc._attachments = {
+          'foo.txt': {
+            content_type: 'text/plain',
+            data: 'Zm9v'
+          }
+        };
+        return db.put(doc);
+      }).then(function (res) {
+        doc._rev = rev2 = res.rev;
+
+        delete doc._attachments;
+        return db.put(doc);
+      }).then(function (res) {
+        doc._rev = rev3 = res.rev;
+
+        return db.getAttachment('a', 'foo.txt', {rev: rev2});
+      }).then(function (blob) {
+        should.exist(blob);
+
+        return Promise.all([
+          db.getAttachment('a', 'foo.txt', {rev: rev1}),
+          db.getAttachment('a', 'foo.txt', {rev: '3-fake'}),
+          db.getAttachment('a', 'foo.txt'),
+          db.getAttachment('a', 'foo.txt', {}),
+          db.getAttachment('a', 'foo.txt', {rev: rev3})
+        ].map(function (promise) {
+          return promise.then(function () {
+            throw new Error('expected an error');
+          }, function (err) {
+            should.exist(err);
+            err.status.should.equal(404);
+          });
+        }));
+      });
+    });
+
```

Comment:
We didn't previously have any tests for `rev` in `getAttachment`. Happily, this just worked&trade; for the local adapters, although it required a fix for the http adapter.